packages/docusaurus-theme-classic/src/theme/NavbarItem/DocSidebarNavbarItem.tsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import DefaultNavbarItem from '@theme/NavbarItem/DefaultNavbarItem';
+import {useLatestVersion, useActiveDocContext} from '@theme/hooks/useDocs';
+import clsx from 'clsx';
+import {getInfimaActiveClassName} from './index';
+import type {Props} from '@theme/NavbarItem/DocSidebarNavbarItem';
+import {useDocsPreferredVersion, uniq} from '@docusaurus/theme-common';
+import type {GlobalDataVersion} from '@docusaurus/plugin-content-docs';
+
+function getSidebarInVersion(versions: GlobalDataVersion[], sidebarId: string) {
+  const allSidebars = versions.map((version) => version.sidebars)[0];
```

Comment:
```suggestion
  const allSidebars = versions.flatMap((version) => version.sidebars);
```

We want to look up for sidebars in all the provided versions, otherwise, it's not useful to provide an array of versions in the first place

Reply:
Hi @slorber, `flatmap` doesn't work I think since `version.sidebars` is an array of `Record<string, GlobalSidebar>` instead of an array of `GlobalSidebar` arrays. I did change it to look for sidebars in all versions in my recent commit.

My question is which version are we getting the sidebar from if there are multiple versions that have the same sidebar?

Reply:
@lmpham1 versions are provided ordered to this method:

`[activeVersion, preferredVersion, latestVersion]`

So sidebars will also be ordered. We'll look for a sidebar in the active version (the one the user is currently browsing) in priority, so this should be fine already: just pick the first sidebar that matches the id